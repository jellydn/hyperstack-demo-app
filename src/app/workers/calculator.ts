import { HyperWorker, queueAs } from 'hyperstack'

@queueAs('calculator')
class Calculator extends HyperWorker {
  async perform({ number }: { number: number }) {
    return Promise.resolve({ number: number * 4 })
  }
}

export { Calculator }
