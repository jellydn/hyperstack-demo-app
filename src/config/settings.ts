// general purpose settings, with no infrastructure loading, no assumptions. keep it simple.
import path from 'path'
export const root = path.join(__dirname, '..')
