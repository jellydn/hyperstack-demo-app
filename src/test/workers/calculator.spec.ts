import { test } from '@hyperstackjs/testing'
import { root } from '../../config/settings'
import { Calculator } from '../../app/workers/calculator'

const { workers } = test(root)

describe('workers', () => {
  describe('downloader', () => {
    workers('should calculate, inprocess', async (_app) => {
      const res = await Calculator.performNow({ number: 30 })
      expect(res).toMatchSnapshot()

      const res2 = await Calculator.performLater({ number: 30 })
      expect(res2).toMatchSnapshot()
    })
  })
})
