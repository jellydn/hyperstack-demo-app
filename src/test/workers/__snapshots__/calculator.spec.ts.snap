// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workers downloader should calculate, inprocess 1`] = `
{
  "number": NaN,
}
`;

exports[`workers downloader should calculate, inprocess 2`] = `
{
  "number": NaN,
}
`;
